pkg/bucketeer/error_mapper.go

```diff
@@ -0,0 +1,71 @@
+package bucketeer
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/api"
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/cache"
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/evaluator"
+	"github.com/bucketeer-io/go-server-sdk/pkg/bucketeer/model"
+)
+
+func MapErrorToReason(err error, isLocalEvaluation bool, featureID string) model.ReasonType {
+	if err == nil {
+		return model.ReasonDefault
+	}
+
+	if errors.Is(err, ErrInvalidUser) || errors.Is(err, ErrUserIDEmpty) {
+		return model.ReasonErrorUserIDNotSpecified
+	}
+	if errors.Is(err, ErrFeatureIDEmpty) {
+		return model.ReasonErrorFeatureFlagIDNotSpecified
+	}
+	if errors.Is(err, ErrUnsupportedType) {
+		return model.ReasonErrorWrongType
+	}
+
+	// Check for response validation errors
+	if errors.Is(err, ErrResponseNil) || errors.Is(err, ErrResponseEvaluationNil) ||
+		errors.Is(err, ErrResponseVariationValueEmpty) || errors.Is(err, ErrResponseFeatureIDMismatch) {
+		return model.ReasonErrorException
+	}
+
+	if isLocalEvaluation {
+		return mapLocalEvaluationError(err)
+	}
+	return mapRemoteEvaluationError(err)
+}
+
+func mapLocalEvaluationError(err error) model.ReasonType {
+	if errors.Is(err, evaluator.ErrEvaluationNotFound) {
+		return model.ReasonErrorFlagNotFound
+	}
+	if errors.Is(err, cache.ErrNotFound) || errors.Is(err, ErrCacheNotReady) ||
+		errors.Is(err, cache.ErrFailedToMarshalProto) || errors.Is(err, cache.ErrProtoMessageNil) ||
+		errors.Is(err, cache.ErrFailedToUnmarshalProto) || errors.Is(err, cache.ErrInvalidType) {
+		return model.ReasonErrorCacheNotFound
+	}
+
+	return model.ReasonErrorException
+}
```

Comment:
For slightly better readability? This is up to you.

Also, maybe these 3, should be reported as `ReasonErrorException`, WDYT?

```go
		errors.Is(err, cache.ErrFailedToMarshalProto) || 
		errors.Is(err, cache.ErrProtoMessageNil) ||
		errors.Is(err, cache.ErrFailedToUnmarshalProto) || 
```

```suggestion
func mapLocalEvaluationError(err error) model.ReasonType {
	if errors.Is(err, evaluator.ErrEvaluationNotFound) {
		return model.ReasonErrorFlagNotFound
	}
	if errors.Is(err, cache.ErrNotFound) || 
		errors.Is(err, ErrCacheNotReady) ||
		errors.Is(err, cache.ErrFailedToMarshalProto) || 
		errors.Is(err, cache.ErrProtoMessageNil) ||
		errors.Is(err, cache.ErrFailedToUnmarshalProto) || 
		errors.Is(err, cache.ErrInvalidType) {
		return model.ReasonErrorCacheNotFound
	}
	return model.ReasonErrorException
}
```